src/vm/excep.cpp

```diff
@@ -7421,6 +7421,24 @@ LONG WINAPI CLRVectoredExceptionHandler(PEXCEPTION_POINTERS pExceptionInfo)
 
     }
 
+    if (pThread)
```

Comment:
I think it would be simpler to have check in just one place in `CLRVectoredExceptionHandlerShim` right after we get the `pThread` there:

```
     // If we cannot obtain a Thread object, then we have no business processing any
     // exceptions on this thread.  Indeed, even checking to see if the faulting
     // address is in JITted code is problematic if we have no Thread object, since
     // this thread will bypass all our locks.
     Thread *pThread = GetThread();

     // Fiber-friendly Vectored Exception Handling
     if (pThread)
     {
         if (on fiber)
         {
             return EXCEPTION_CONTINUE_SEARCH;
         }
     }
```

Reply:
I am not that familiar with fibers, but I wonder - would GetCurrentFiber() returning null / not null work as an indicator of running on a fiber here?

Reply:
That does not work. `GetCurrentFiber()` returns non-null even on regular threads - regular threads are fibers too.

Reply:
It's the same for `IsThreadAFiber()`. There is no concept of "main fiber", so checking the stack-base seems to be the only way to detect if the fiber changed.
But, if the Thread-object recorded the current Fiber on initialization with `GetCurrentFiber()` then we could use it to detect if it changed.

Reply:
I think the stack base comparison is ok. We don't need to add extra info to the Thread if this is sufficient.